alpakaConfig.cmake

```diff
@@ -678,16 +686,14 @@ IF(ALPAKA_ACC_GPU_CUDA_ENABLE)
                 # avoids warnings on host-device signature of 'std::__shared_count<>'
                 IF(CUDA_VERSION EQUAL 10.0)
                     LIST(APPEND CUDA_NVCC_FLAGS -Xcudafe --diag_suppress=2905)
-                ENDIF()
-
-                # avoids warnings on host-device signature of 'std::__shared_count<>'
-                IF(CUDA_VERSION EQUAL 10.1)
+                ELSEIF(CUDA_VERSION EQUAL 10.1)
                     LIST(APPEND CUDA_NVCC_FLAGS -Xcudafe --diag_suppress=2912)
+                ELSEIF(CUDA_VERSION EQUAL 10.2)
+                    LIST(APPEND CUDA_NVCC_FLAGS -Xcudafe --diag_suppress=2976)
                 ENDIF()
 
-                # avoids warnings on host-device signature of 'std::__shared_count<>'
                 IF(CUDA_VERSION EQUAL 10.2)
-                    LIST(APPEND CUDA_NVCC_FLAGS -Xcudafe --diag_suppress=2976)
+                    LIST(APPEND CUDA_NVCC_FLAGS -Werror=all-warnings)
```

Comment:
`-Werror` of any kind should never be shipped to users. Activate only in CI, but not in CMake default flags. We cannot assume downstream usage and will break a lot when setting this.

Reply:
You are right, I will add this to the tests only.